src/slickml/selection/_xgboost.py

```diff
@@ -0,0 +1,1139 @@
+                    self._cvr,
+                    self._evals_result,
+                    _feature_gain,
+                    _X_train,
+                    _y_train,
+                    _X_test,
+                    _y_test,
+                )
+
+                ijk += 1
+                gc.collect()
+
+            # print internal metrics results
+            # TODO(amir): replace print with logging
+            print(
+                str(Colors.BOLD)
+                + "*-*-* "
+                + str(Colors.GREEN)
+                + f"Internal {self.n_splits}-Folds CV:"
+                + str(Colors.END)
+                + str(Colors.BOLD)
+                + " -*-*- "
+                + str(Colors.F_Red)
+                + f"Train {self.metrics.upper()}"
+                + " = "
+                + f"{np.mean(int_cv_train2):.3f}"
+                + " +/- "
+                + f"{np.std(int_cv_train2):.3f}"
+                + str(Colors.END)
+                + str(Colors.BOLD)
+                + " -*-*- "
+                + str(Colors.F_Blue)
+                + f"Test {self.metrics.upper()}"
+                + " = "
+                + f"{np.mean(int_cv_test2):.3f}"
+                + " +/- "
+                + f"{np.std(int_cv_test2):.3f}"
+                + str(Colors.END)
+                + str(Colors.BOLD)
+                + " *-*-*",
+            )
+
+            #  print external eval_metric results
+            # TODO(amir): replace print with logging
+            print(
+                str(Colors.BOLD)
+                + "*-*-* "
+                + str(Colors.GREEN)
+                + f"External {self.n_splits}-Folds CV:"
+                + str(Colors.END)
+                + str(Colors.BOLD)
+                + " -*-*- "
+                + str(Colors.F_Red)
+                + f"Train {self.params['eval_metric'].upper()}"
+                + " = "
+                + f"{np.mean(ext_cv_train2):.3f}"
+                + " +/- "
+                + f"{np.std(ext_cv_train2):.3f}"
+                + str(Colors.END)
+                + str(Colors.BOLD)
+                + " -*-*- "
+                + str(Colors.F_Blue)
+                + f"Test {self.params['eval_metric'].upper()}"
+                + " = "
+                + f"{np.mean(ext_cv_test2):.3f}"
+                + " +/- "
+                + f"{np.std(ext_cv_test2):.3f}"
+                + str(Colors.END)
+                + str(Colors.BOLD)
+                + " *-*-*\n",
+            )
+
+            # free memory here at iteration
+            del (
+                int_cv_train2,
+                int_cv_test2,
+                ext_cv_train2,
+                ext_cv_test2,
+                _X_permuted,
+                _X_permuted_values,
+                _columns,
+            )
+            gc.collect()
+
+        self.plotting_cv_ = self._plotting_cv()
+        self.feature_frequency_ = self._freq()
+
+        return None
+
+    def plot_frequency(
+        self,
+        *,
```

Comment:
We need to make `*` required for all named args in the class methods. We did have them required in the `upstream` viz functions; not for the downstream ones tho.